app/controllers/api/v1/classes_controller.rb

```diff
@@ -98,34 +98,52 @@ def log_links
 
   private
 
-
-
   def render_info(clazz)
     state = nil
     if school = clazz.school
       state = school.state
     end
 
     render :json => {
-      :uri => url_for(clazz),
+      :uri => api_v1_class_url(clazz.id),
       :name => clazz.name,
       :state => state,
       :class_hash => clazz.class_hash,
-      :teachers => clazz.teachers.includes(:user).map{|teacher|
+      :class_word => clazz.class_word,
+      :edit_path => edit_portal_clazz_path(clazz),
+      :assign_materials_path => configured_search_path,
```

Comment:
@scytacki, re your comment about ITSI - I'm already passing this to Portal Pages. I guess I could check theme here and just return `"/itsi?assign_to_class=#{clazz.id}"` if it's ITSI.

Re fake id = 999, it won't be necessary here, as path will be resolved on Portal side, but I've also started adding function helpers to Portal.API_V1, like this one:
https://github.com/concord-consortium/rigse/blob/22a6f0389185da871fb45a20524d51e63786f720/app/views/dynamic_scripts/_api_paths.html.haml#L9-L12
Following convetion that the name is the same, but camelCase and accepting IDs as params.

Reply:
Yes that approach sounds good. It might one of the only places where we have theme'd logic directly in a controller, but I don't see another way to do it.